src/main/kotlin/com/wink/dbse/command/economy/Profile.kt

```diff
@@ -0,0 +1,75 @@
+package com.wink.dbse.command.economy
+
+import com.jagrosh.jdautilities.command.Command
+import com.jagrosh.jdautilities.command.CommandEvent
+import com.wink.dbse.entity.BangEntity
+import com.wink.dbse.entity.UserEntity
+import com.wink.dbse.entity.game.BangResult
+import com.wink.dbse.extension.safeName
+import com.wink.dbse.repository.BangRepository
+import com.wink.dbse.repository.UserRepository
+import com.wink.dbse.service.Messenger
+import net.dv8tion.jda.api.EmbedBuilder
+import org.springframework.stereotype.Component
+import java.awt.Color
+import kotlin.math.roundToInt
+
+@Component
+class Profile(
+    private val messenger: Messenger,
+    private val userRepository: UserRepository,
+    private val bangRepository: BangRepository
+) : Command() {
+
+    init {
+        name = "profile"
+        help = "shows your profile information"
+    }
+
+    private val eb = EmbedBuilder()
+
+    override fun execute(event: CommandEvent) {
+        eb.clear()
+        setHeading(event)
+        generateBangInfo(bangRepository.findBangEntitiesByUserId(event.author.idLong))
+        setUserInfo(userRepository.findById(event.author.idLong).get())
+        messenger.sendMessage(event.channel, eb.build())
+    }
+
+    private fun setUserInfo(user: UserEntity) {
+        eb.addField("Streak", "${user.bangStreak}" + if(user.bangStreak > 0)" :fire:" else "", true)
+        eb.addField("Karma","${user.upVotes - user.downVotes}",false)
+        eb.addField("Wallet","${user.wallet} gc",true)
+    }
+
+    private fun generateBangInfo(bangs: List<BangEntity>) {
+        var jams = 0
+        var deaths = 0
+        //TODO: remove for the new database option
```

Comment:
still needs to be adjusted but we need to figure out how to do the repository method or ill have to update the user each time they bang